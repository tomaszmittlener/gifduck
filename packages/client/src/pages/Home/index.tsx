import React, { FunctionComponent, useEffect, useState } from 'react'
import styled from 'styled-components'
import { useLocation } from 'react-router-dom'
import { SearchQueryKeys, ImageData, SearchQuery } from '@gifduck/types/imagesService'

import { getQuery } from 'utilities/query'
import imagesService from 'services/imagesService'
import Header from 'containers/Header'
import Footer from 'containers/Footer'
import SearchResults from 'containers/SearchResults'

interface HomePageProps {
  className?: string
  testId?: string
}

const PageWrapper = styled.div`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.colors.grays.white};
`

const Main = styled.main`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: center;
  min-height: 50vh;
  padding: 0 0 ${({ theme }) => theme.ms(15)} 0;
`

const HomePage: FunctionComponent<HomePageProps> = ({ className, testId }) => {
  const [results, storeResults] = useState<ImageData[]>([])
  const { search } = useLocation()

  const getData = async () => {
    try {
      const searchText = getQuery(search)[SearchQueryKeys.searchText]
      const searchQuery: SearchQuery = {
        [SearchQueryKeys.searchText]: searchText ? String(searchText) : '',
      }
      const { data } = await imagesService.getImages(searchQuery)
      storeResults(data.results)
    } catch (e) {
      throw new Error(e)
    }
  }

  useEffect(() => {
    getData()
  }, [search])

  return (
    <PageWrapper data-testid={testId} className={className}>
      <Header />
      <Main>
        <SearchResults results={results} />
      </Main>
      <Footer />
    </PageWrapper>
  )
}

export default HomePage
