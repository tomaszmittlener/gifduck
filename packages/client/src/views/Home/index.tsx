import React, { FunctionComponent, useEffect, useState } from 'react'
import styled from 'styled-components'
import { useLocation } from 'react-router-dom'
import { ImageData, SearchQuery, SearchQueryKeys } from '@gifduck/types/imagesService'

import { getQuery } from 'common/queryString'
import imagesService from 'services/imagesService'
import Header from 'containers/Header'
import Footer from 'containers/Footer'
import SearchResults from 'containers/SearchResults'
import { FetchStatus } from 'common/types'
import scrollToTop from 'common/scrollToTop'

interface HomePageProps {
  className?: string
  testId?: string
}

const PageWrapper = styled.div`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.colors.grays.white};
`

const Main = styled.main`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: center;
  min-height: 50vh;
  padding: 0 0 ${({ theme }) => theme.ms(15)} 0;
`

const HomePage: FunctionComponent<HomePageProps> = ({ className, testId }) => {
  const [results, storeResults] = useState<ImageData[]>([])
  const [fetchStatus, updateFetchStatus] = useState<FetchStatus>(FetchStatus.Loading)

  const { search } = useLocation()

  const getData = async () => {
    try {
      updateFetchStatus(FetchStatus.Loading)

      const searchText = getQuery(search)[SearchQueryKeys.searchText]
      const searchQuery: SearchQuery = {
        [SearchQueryKeys.searchText]: searchText ? String(searchText) : '',
      }
      const { data } = await imagesService.getImages(searchQuery)
      storeResults(data.results)
      updateFetchStatus(FetchStatus.Loaded)

      scrollToTop()
    } catch (e) {
      updateFetchStatus(FetchStatus.Error)

      throw new Error(e)
    }
  }

  useEffect(() => {
    getData()
  }, [search])

  return (
    <PageWrapper data-testid={testId} className={className}>
      <Header isLoading={fetchStatus === FetchStatus.Loading} />
      <Main>
        <SearchResults results={results} />
      </Main>
      <Footer />
    </PageWrapper>
  )
}

export default HomePage
