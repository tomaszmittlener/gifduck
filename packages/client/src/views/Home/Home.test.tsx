import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import Theme from 'styles/theme'

import Home from './'
import { MemoryRouter } from 'react-router-dom'

// const defaultProps = {
//   className: 'class-name',
//   testId: 'test-id',
// }

afterEach(cleanup)

describe('SearchResults', () => {
  test('renders correctly', async () => {
    const { container } = render(
      <ThemeProvider theme={Theme}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </ThemeProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  // test('passes className to container', async () => {
  //   const container = render(
  //     <ThemeProvider theme={Theme}>
  //       <MemoryRouter>
  //         <Home {...defaultProps} />
  //       </MemoryRouter>
  //     </ThemeProvider>,
  //   )
  //
  //   expect(container.getByTestId(defaultProps.testId)).toHaveClass(defaultProps.className)
  // })
})
