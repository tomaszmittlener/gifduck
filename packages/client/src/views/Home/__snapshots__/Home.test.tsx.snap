// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
.c3 {
  font-size: 2.2806973457336426rem;
  font-family: Pacifico,sans-serif;
  color: #F2C94C;
  line-height: 2;
  width: auto;
  -webkit-animation: dGzoaH 1s ease-out infinite;
  animation: dGzoaH 1s ease-out infinite;
}

.c1 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  width: 100%;
  z-index: 1500;
  background-color: #343339;
}

.c2 {
  padding: 0 1rem;
  grid-column-gap: 0.49327018427257213rem;
  display: grid;
  grid-template-columns: repeat(8,1fr);
  width: 100%;
  max-width: 1440px;
  margin: 0 auto;
}

.c5 {
  grid-column: 1 / 9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 0 1.601806640625rem 0;
}

.c7 {
  height: 3.247321025468409rem;
  width: 3.247321025468409rem;
  border: none;
  background-color: #ECE9DE;
  padding: 0;
  margin: 0;
  cursor: pointer;
  border-radius: 0 10px 10px 0;
}

.c6 {
  border: none;
  line-height: 1.75;
  height: 3.247321025468409rem;
  color: #5D899E;
  background-color: #ECE9DE;
  font-size: 1rem;
  font-family: Open Sans,sans-serif;
  padding: 0 1rem;
  width: 100%;
  border-radius: 10px 0 0 10px;
}

.c4 {
  padding: 0.7901234567901234rem 0;
  grid-column: 1 / 9;
  justify-self: center;
}

.c11 {
  margin: auto 0 0 0;
  width: 100%;
  min-height: 19.00260151070504rem;
  background-color: #F2C94C;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1.423828125rem 0px;
}

.c12 {
  color: #706155;
  font-family: Pacifico,sans-serif;
  font-size: 0.7901234567901234rem;
}

.c9 {
  padding: 0 1rem;
  grid-column-gap: 0.49327018427257213rem;
  display: grid;
  grid-template-columns: repeat(8,1fr);
  width: 100%;
  max-width: 1440px;
  height: 100%;
}

.c10 {
  grid-column: 3 / 7;
  justify-self: center;
  font-family: Open Sans,sans-serif;
  color: #343339;
  font-size: 3.247321025468409rem;
  text-align: center;
}

.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #ECE9DE;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 50vh;
  padding: 0 0 5.851777930691043rem 0;
}

@media screen and (min-width:768px) {
  .c2 {
    padding: 0 1.423828125rem;
  }
}

@media screen and (min-width:1024px) {
  .c2 {
    padding: 0 2.0272865295410156rem;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    grid-column-gap: 1rem;
  }
}

@media screen and (min-width:768px) {
  .c4 {
    justify-self: left;
    grid-column: 1 / 4;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    padding: 0 1.423828125rem;
  }
}

@media screen and (min-width:1024px) {
  .c9 {
    padding: 0 2.0272865295410156rem;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    grid-column-gap: 1rem;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <label
          class="c3 c4"
        >
          üê§gifduck
        </label>
        <form
          class="c5"
        >
          <input
            class="c6"
            name="search"
            placeholder="Search for Images..."
            type="text"
          />
          <button
            class="c7"
          >
            üîé
          </button>
        </form>
      </div>
    </header>
    <main
      class="c8"
    >
      <section
        class="c9"
      >
        <h3
          class="c10"
        >
          Type in whats on your mind!
        </h3>
      </section>
    </main>
    <footer
      class="c11"
    >
      <span
        class="c12"
      >
        @gifduck
      </span>
    </footer>
  </div>
</div>
`;
